E0128 needs to be updated to new format

From: src/test/compile-fail/E0128.rs

Error E0128 needs a span_label, updating it from:

```
error[E0128]: type parameters with a default cannot use forward declared identifiers
  --> src/test/compile-fail/E0128.rs:11:14
   |
11 | struct Foo<T=U, U=()> { //~ ERROR E0128
   |              ^
```

To:

```
error[E0128]: type parameters with a default cannot use forward declared identifiers
  --> src/test/compile-fail/E0128.rs:11:14
   |
11 | struct Foo<T=U, U=()> { //~ ERROR E0128
   |              ^ defaulted type parameters cannot be forward declared
```

--- a/src/librustc_typeck/collect.rs
+++ b/src/librustc_typeck/collect.rs
@@ -1901,9 +1901,12 @@ fn convert_default_type_parameter<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
     for leaf_ty in ty.walk() {
         if let ty::TyParam(p) = leaf_ty.sty {
             if p.space == space && p.idx >= index {
-                span_err!(ccx.tcx.sess, path.span, E0128,
-                          "type parameters with a default cannot use \
-                           forward declared identifiers");
+                struct_span_err!(ccx.tcx.sess, path.span, E0128,
+                                 "type parameters with a default cannot use \
+                                 forward declared identifiers")
+                    .span_label(path.span, &format!("defaulted type parameters \
+                                                    cannot be forward declared"))
+                    .emit();
 
                 return ccx.tcx.types.err
             }

--- a/src/test/compile-fail/E0128.rs
+++ b/src/test/compile-fail/E0128.rs
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 struct Foo<T=U, U=()> { //~ ERROR E0128
+                        //~| NOTE defaulted type parameters cannot be forward declared
     field1: T,
     field2: U,
 }

--- a/src/test/compile-fail/issue-18183.rs
+++ b/src/test/compile-fail/issue-18183.rs
@@ -9,5 +9,6 @@
 // except according to those terms.
 
 pub struct Foo<Bar=Bar>; //~ ERROR E0128
+                         //~| NOTE defaulted type parameters cannot be forward declared
 pub struct Baz(Foo);
 fn main() {}

